ingestion/src/metadata/utils/secrets_manager.py

```diff
@@ -36,24 +46,40 @@
     "ServiceConnectionType", ServiceConnection.__fields__["__root__"].type_
 )
 
+AuthProviderClientType = NewType(
+    "AuthProviderClientType", OpenMetadataConnection.__fields__["securityConfig"].type_
+)
+
+AUTH_PROVIDER_MAPPING: Dict[AuthProvider, AuthProviderClientType] = {
+    AuthProvider.google: googleSSOClientConfig.GoogleSSOClientConfig,
+    AuthProvider.okta: oktaSSOClientConfig.OktaSSOClientConfig,
+    AuthProvider.auth0: auth0SSOClientConfig.Auth0SSOClientConfig,
+    AuthProvider.azure: azureSSOClientConfig.AzureSSOClientConfig,
+    AuthProvider.custom_oidc: customOidcSSOClientConfig.CustomOIDCSSOClientConfig,
+    AuthProvider.openmetadata: openMetadataJWTClientConfig.OpenMetadataJWTClientConfig,
+}
+
 
 class SecretsManager(metaclass=Singleton):
     @abstractmethod
     def add_service_config_connection(
         self,
         service: ServiceConnectionType,
         service_type: str,
-    ) -> ServiceConnectionType:
+    ):
+        pass
+
+    @abstractmethod
+    def add_auth_provider_security_config(self, config: OpenMetadataConnection):
         pass
 
     @staticmethod
     def to_service_simple(service_type: str) -> str:
         return service_type.replace("Service", "").lower()
 
-    def build_secret_id(
-        self, service_type: str, service_connection_type: str, service_name: str
-    ) -> str:
-        return f"openmetadata-{self.to_service_simple(service_type).lower()}-{service_connection_type.lower()}-{service_name.lower()}"
+    def build_secret_id(self, parameters: List[str]) -> str:
```

Comment:
I believe we can use `*args` here instead of passing the parameters as a list. It might better capture that we need a varying number of elements.

When building it would be a bit cleaner as well IMO:

```python
secret_id = self.build_secret_id(
            "auth-provider", config.authProvider.value
        )
```